src/jabs/ui/central_widget.py

```diff
@@ -590,11 +595,57 @@ def _train_button_clicked(self) -> None:
 
     def _training_thread_complete(self) -> None:
         """enable classify button once the training is complete"""
-        self._progress_dialog.close()
-        self._progress_dialog.deleteLater()
-        self._progress_dialog = None
+        self._cleanup_training_thread()
+        self._cleanup_progress_dialog()
         self.status_message.emit("Training Complete", 3000)
-        self._controls.classify_button_set_enabled(True)
+        self._controls.classify_button_enabled = True
+
+    def _training_thread_error_callback(self, error: Exception) -> None:
+        """handle an error in the training thread"""
+        self._print_exception(error)
+        self._cleanup_training_thread()
+        self._cleanup_progress_dialog()
+        self.status_message.emit("Training Failed", 3000)
+        self._controls.classify_button_enabled = False
+        QtWidgets.QMessageBox.critical(
+            self, "Error", f"An exception occurred during training:\n{error}"
+        )
+
+    def _classify_thread_error_callback(self, error: Exception) -> None:
+        """handle an error in the classification thread"""
+        self._print_exception(error)
+        self._cleanup_training_thread()
```

Comment:
oops. this is wrong -- need to create a _cleanup_classify_thread method to call here